server/spec/content_access_spec.rb

```diff
@@ -726,23 +722,42 @@ def regenerate_cert_test(&updater)
   end
 
   it 'should regenerate SCA cert when content changes affect content view' do
-    regenerate_cert_test do
+    client = consumer_client(@user, @consumername, type=:system, username=nil, facts= {'system.certificate_version' => '3.3'})
+
+    regenerate_cert_test(client) do
       @cp.update_content(@owner['key'], @content['id'], { 'content_url' => '/updated/path' })
     end
   end
 
   it 'should regenerate SCA cert when product changes affect content view' do
-    regenerate_cert_test do
+    client = consumer_client(@user, @consumername, type=:system, username=nil, facts= {'system.certificate_version' => '3.3'})
+
+    regenerate_cert_test(client) do
       @cp.remove_content_from_product(@owner['key'], @product['id'], @content['id'])
     end
   end
 
   it 'should regenerate SCA cert when pool changes affect content view' do
-    regenerate_cert_test do
+    client = consumer_client(@user, @consumername, type=:system, username=nil, facts= {'system.certificate_version' => '3.3'})
+
+    regenerate_cert_test(client) do
       @pool['start_date'] = (DateTime.now + 1)
       @cp.update_pool(@owner['key'], @pool)
     end
   end
 
+  it 'should regenerate SCA cert when environment content changes' do
```

Comment:
This test doesn't fail when I run it against a candlepin without the fix. I am also having trouble reproducing the issue manually (either doing unregister/register, or subman refresh)

Reply:
I have tried reproducing manually with both using an environment, and without an environment, but still cannot reproduce

Reply:
And it shouldn't. This test was the first one I wrote to attempt to reproduce it after the initial report. It doesn't hit the issue because the root cause here is that a consumer with a non-null environment needs to have an existing cert that expires naturally which would then be regenerated. The spec tests cannot verify this path without priming/modifying the database, so this is done in the unit tests which were also added as part of this PR.